src/Unity.Mathematics/random.cs

```diff
@@ -27,6 +27,49 @@ public Random(uint seed)
             NextState();
         }
 
+        /// <summary>
+        /// Constructs a <see cref="Random"/> instance with an index that gets hashed.  The index must not be uint.MaxValue.
+        /// </summary>
+        /// <remarks>
+        /// Use this function when you expect to create several Random instances in a loop.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// for (uint i = 0; i &lt; 4096; ++i)
+        /// {
+        ///     Random rand = Random.CreateFromIndex(i);
+        ///
+        ///     // Random numbers drawn from loop iteration j will be very different
+        ///     // from every other loop iteration k.
+        ///     rand.NextUInt();
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="index">An index that will be hashed for Random creation.  Must not be uint.MaxValue.</param>
+        /// <returns><see cref="Random"/> created from an index.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Random CreateFromIndex(uint index)
+        {
+            CheckIndexForHash(index);
+
+            // Wang hash (below) will hash 61 to zero so choose a different index to hash to zero.
```

Comment:
Would it be worthwhile to move the Wang hash to an internal function? That might make it easier to add a unit test that WangHash(61) == 0

Reply:
Great idea, thanks